### PR TITLE
[JIT] Fix bug in aggregate code generation

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@Kixiron I added a couple of messages to assert and unwrap to make it easier to diagnose issues I stumbled into.

File: crates/dataflow-jit/src/codegen/layout.rs [hunk 1]
```diff
@@ -436,7 +436,9 @@ impl NativeLayout {
     ///
     /// Panics if `column` isn't nullable
     pub fn nullability_of(&self, column: usize) -> (BitSetType, u32, u8) {
-        self.bitsets[column].unwrap()
+        self.bitsets[column].unwrap_or_else(
+	    || panic!("{:?}: checking nullability of non-nullable column {:?}",
+		      self.bitsets, column))
```
Comment (Kixiron) [inline crates/dataflow-jit/src/codegen/layout.rs:441]:
```suggestion
        self.bitsets[column].unwrap_or_else(|| {
            panic!("{:?}: checking nullability of non-nullable column {:?}", self.bitsets, column)
        })
```

File: crates/dataflow-jit/src/codegen/mod.rs [hunk 2]
```diff
@@ -1614,8 +1614,10 @@ impl<'a> CodegenCtx<'a> {
     fn binary_op(&mut self, expr_id: ExprId, binop: &BinaryOp, builder: &mut FunctionBuilder<'_>) {
         let (lhs, rhs) = (self.exprs[&binop.lhs()], self.exprs[&binop.rhs()]);
         let (lhs_ty, rhs_ty) = (self.expr_types[&binop.lhs()], self.expr_types[&binop.rhs()]);
-        debug_assert_eq!(builder.value_type(lhs), builder.value_type(rhs));
-        debug_assert_eq!(lhs_ty, rhs_ty);
+        debug_assert_eq!(builder.value_type(lhs), builder.value_type(rhs),
+			 "Operands of binary operation {:?} have different types", binop);
+        debug_assert_eq!(lhs_ty, rhs_ty,
+			 "Operands of binary operation {:?} have different types", binop);
```
Comment (Kixiron) [inline crates/dataflow-jit/src/codegen/mod.rs:1620]:
```suggestion
        debug_assert_eq!(
            builder.value_type(lhs),
            builder.value_type(rhs),
            "Operands of binary operation {:?} have different types",
            binop,
        );
        debug_assert_eq!(
            lhs_ty,
            rhs_ty,
            "Operands of binary operation {:?} have different types",
            binop,
        );
```

**Review by Kixiron — APPROVED** [review]

File: crates/dataflow-jit/src/codegen/layout.rs [hunk 3]
```diff
@@ -436,7 +436,10 @@ impl NativeLayout {
     ///
     /// Panics if `column` isn't nullable
     pub fn nullability_of(&self, column: usize) -> (BitSetType, u32, u8) {
-        self.bitsets[column].unwrap()
+        self.bitsets[column].unwrap_or_else(|| {
+	    panic!("{:?}: checking nullability of non-nullable column {:?}",
+		   self.bitsets, column)
```
Comment (Kixiron) [inline crates/dataflow-jit/src/codegen/layout.rs:441]:
```suggestion
        panic!(
	        "{:?}: checking nullability of non-nullable column {:?}",
		    self.bitsets,
            column,
        )
```

**Pull request merged**